languages/go/exercises/concept/using-map/.docs/after.md

```diff
@@ -0,0 +1,35 @@
+`map` in go is a key-value type. In other programming language, you might know map as `dict` or associative array. The value of an uninitialized map is `nil`, and you can define map as follows
+
+`make(map[string]int)`
+
+or
+
+```go
+  var foo map[string]int
+  foo = make(map[string]int)
+```
+
+> Here `string` represent the key type of map, and `int` represent the element type of map. You can change them to whatever valid golang type.
+
+To store a value in a map, you can use `=` operator like
+
+`foo["bar"] = 42`
+
+`map` is indexed by its key, therefore, you are required to use different key if you want to store a different data.
+
+To get value from a map, you can use
+
+`foo["bar"]`
+
+Retrieving inexistent key from map will cause a runtime error, and to check if such a key exists you can do
```

Comment:
```suggestion
Retrieving a value using a non-existent key will result in a runtime error. One can use the optional second return value to check if the key did exist:
```

Reply:
This is statement is actually false and should be:
```suggestion
Retrieving a value using a non-existent key will return the zero value for the values type. It will not error as you are used to from many other languages. To check if the key did exist, one can use the optional second return value:
```

Reply:
This link could be added here: https://golang.org/ref/spec#The_zero_value

Reply:
I like the suggestion!